static_report/src/components/ComparisonReport/CRModal.tsx

```diff
@@ -0,0 +1,105 @@
+import {
+  Button,
+  Table,
+  TableContainer,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+  Text,
+  Modal,
+  ModalHeader,
+  ModalOverlay,
+  ModalContent,
+  ModalCloseButton,
+  ModalBody,
+  ModalFooter,
+  type UseDisclosureReturn,
+} from '@chakra-ui/react';
+
+import { TestStatus } from '../shared/TestStatus';
+import { extractExpectedOrActual } from '../../utils';
+
+interface Props extends UseDisclosureReturn {
+  data?: any;
+  type?: 'piperider' | 'dbt';
+}
+
+export function CRModal({
+  data,
+  type = 'piperider',
+  isOpen,
+  onClose,
+  ...props
+}: Props) {
+  return (
+    <Modal
+      {...props}
+      isOpen={isOpen}
+      size="2xl"
+      onClose={() => {
+        onClose();
+      }}
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>
+          <Text title={data?.name} noOfLines={1} maxWidth="calc(100% - 50px)">
+            {data?.name}
+          </Text>
+        </ModalHeader>
+        <ModalCloseButton />
+        <ModalBody>
+          <TableContainer>
+            <Table variant="simple">
+              <Thead>
+                <Tr>
+                  <Th />
+                  <Th>Status</Th>
+                  {type === 'piperider' && <Th>Expected</Th>}
```

Comment:
i think since there's alot of `'piperider' || 'dbt'` logic going on here, 
I suggest we break into two separate table blocks, to better represent each view.